File: LeetCode/Easy/455-Assign_Cookies/455-AssignCookies.ex
defmodule Solution do
  @spec find_content_children(g :: [integer], s :: [integer]) :: integer
  def find_content_children(g, s) do
    sorted_g = Enum.sort(g)
    sorted_s = Enum.sort(s)

    match_kid(sorted_g, sorted_s, 0)
  end

  defp match_kid([], _size, count), do: count

  defp match_kid(_greedy, [], count), do: count

  defp match_kid([curr_greedy | rest_g], [curr_size | rest_s], count) do
    if curr_greedy <= curr_size do
      match_kid(rest_g, rest_s, count + 1)
    else
      match_kid([curr_greedy | rest_g], rest_s, count)
    end
  end
end

File: LeetCode/Easy/455-Assign_Cookies/455-AssignCookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function (g, s) {
    let n = g.length;
    let m = s.length;

    g.sort((a, b) => a - b);
    s.sort((a, b) => a - b);

    let j = 0;
    let count = 0;

    for (let i = 0; i < n; i++) {
        while (j < m && g[i] > s[j])
            j++;

        if (j < m) {
            count++;
            j++;
        } else
            break;
    }

    return count;
};